server/src/main/java/org/apache/cassandra/sidecar/tasks/CdcRawDirectorySpaceCleaner.java

```diff
@@ -0,0 +1,437 @@
+                                       CdcConfiguration cdcConfiguration,
+                                       InstanceMetadata instanceMetadata,
+                                       SidecarMetrics metrics)
+    {
+        this.timeProvider = timeProvider;
+        this.systemViewsDatabaseAccessor = systemViewsDatabaseAccessor;
+        this.cdcConfiguration = cdcConfiguration;
+        this.instanceMetadata = instanceMetadata;
+        this.cdcMetrics = metrics.server().cdc();
+    }
+
+    @Override
+    public DurationSpec delay()
+    {
+        return cdcConfiguration.cdcRawDirectorySpaceCleanerFrequency();
+    }
+
+    @Override
+    public void execute(Promise<Void> promise)
+    {
+        try
+        {
+            routineCleanUp();
+            promise.tryComplete();
+        }
+        catch (Throwable t)
+        {
+            LOGGER.warn("Failed to perform routine clean-up of cdc_raw directory", t);
+            cdcMetrics.cdcRawCleanerFailed.metric.update(1L);
+            promise.fail(t);
+        }
+    }
+
+    /**
+     * @return true if we need to refresh the cached `cdc_total_space` value.
+     */
+    protected boolean shouldRefreshCachedMaxUsage()
+    {
+        return maxUsageLastReadNanos == null ||
+               (System.nanoTime() - maxUsageLastReadNanos) >= cdcConfiguration.cacheMaxUsage().toNanos();
+    }
+
+    protected long maxUsage()
+    {
+        if (!shouldRefreshCachedMaxUsage())
+        {
+            return Objects.requireNonNull(maxUsageBytes, "maxUsageBytes cannot be null if maxUsageLastReadNanos is non-null");
+        }
+
+        try
+        {
+            Long newValue = systemViewsDatabaseAccessor.getCdcTotalSpaceSetting();
+            if (newValue != null)
+            {
+                if (!newValue.equals(maxUsageBytes))
+                {
+                    LOGGER.info("Change in cdc_total_space from system_views.settings prev={} latest={}", maxUsageBytes, newValue);
+                    this.maxUsageBytes = newValue;
+                    this.maxUsageLastReadNanos = System.nanoTime();
+                    return newValue;
+                }
+            }
+        }
+        catch (SchemaUnavailableException e)
+        {
+            LOGGER.debug("Could not read cdc_total_space from system_views.settings", e);
+        }
+        catch (Throwable t)
+        {
+            LOGGER.warn("Error reading cdc_total_space from system_views.settings", t);
+        }
+
+        LOGGER.warn("Could not read cdc_total_space from system_views.settings, falling back to props");
+        return cdcConfiguration.fallbackCdcRawDirectoryMaxSizeBytes();
+    }
+
+    protected void routineCleanUp()
+    {
+        if (!cdcConfiguration.enableCdcRawDirectoryRoutineCleanUp())
+        {
+            LOGGER.debug("Skipping CdcRawDirectorySpaceCleaner: feature is disabled ");
+            return;
+        }
```

Comment:
Instead of having this check here, why not just prevent the task from being scheduled by overriding `scheduleDecision` method? Something like:
```
@Override
public ScheduleDecision scheduleDecision()
{
    return cdcConfiguration.enableCdcRawDirectoryRoutineCleanUp() ? ScheduleDecision.EXECUTE : ScheduleDecision.SKIP;
}
```